app/src/main/java/no/usn/mob3000/data/source/remote/social/FriendRequestDataSource.kt

```diff
@@ -0,0 +1,128 @@
+package no.usn.mob3000.data.source.remote.social
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.postgrest.from
+import io.github.jan.supabase.postgrest.result.PostgrestResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import no.usn.mob3000.data.model.social.FriendRequestsDto
+import no.usn.mob3000.data.model.social.FriendsDto
+import no.usn.mob3000.data.network.SupabaseClientWrapper
+/**
+ *
+ * A data source class responsible for handling friend request-related calls via Supabase.
+ *
+ * @param supabaseClient The Supabase client for interacting with the backend.
+ * @constructor Creates an instance of [FriendRequestDataSource].
+ * @author Husseinabdulameer11, 258030
+ * @created 2024-11-13
+ */
+
+ class FriendRequestDataSource (private val supabaseClient: SupabaseClient = SupabaseClientWrapper.getClient()){
```

Comment:
Dangling KDoc. Additionally, the parameter list and the class's opening bracket should be separated. The class property should also be indented.